foundation/foundation-mda/src/main/resources/templates/deployment/argocd/v2/spark.infrastructure.yaml.vm

```diff
@@ -6,6 +6,15 @@ metadata:
   finalizers:
     - resources-finalizer.argocd.argoproj.io
 spec:
+  ignoreDifferences:
+    - group: ""
+      kind: ConfigMap
+      managedFieldsManagers:
+        - argocd-controller
```

Comment:
Q: Shouldn't the `managedFieldsManagers` include the configuration store controller? How is the config webhook marking injected fields as managed?

Reply:
![Screenshot 2024-12-19 at 3 52 35 PM](https://github.com/user-attachments/assets/850535e6-c5ff-484e-825a-68d90c69de58)
Upon inspection of local argoCD live manifest, those seems to be managed by argocd-controller

Reply:
Can you connect the dots for me? Why would fields modified by the config store show up as managed by argocd-controller? That seems incorrect on it's face, since the config store could be used outside of the context of ArgoCD. In that case, I'd expect you'd still want the fields to be flagged as managed.

Reply:
I think I could mark only `data: metastore-site.xml` to be  managed by something called `configstore-controller`
but actual issue happens in Statefulset in metastoreDB, as far as I have investigated there's not really a way to create custom managed fields unless we actually untar the mysql and change it from there.

Reply:
But argocd is showing up as the manager for those fields? ArgoCD has no explicit knowledge of the charts being deployed. If we would have to untar the mysql chart to set the config store as the manager, how is ArgoCD showing up as the manager?

Reply:
Correct me if I am wrong 

I couldn't find exact point where ArgoCD has explicit check on chart being deployed but upon investigation of argoCD codebase, I think this is default manager while doing server side apply syncing process.

Details of code can be found here:

ArgoCD controller constant : https://github.com/argoproj/argo-cd/blob/ac8e3f01e9431e1c29a5b0e8af7ac2a962a42e19/common/common.go#L110

Sync Logic: https://github.com/argoproj/argo-cd/blob/ac8e3f01e9431e1c29a5b0e8af7ac2a962a42e19/controller/sync.go#L380


Upon sync, I think it sets default manager.

Reply:
Discussed offline, it appears there's not good way to set the manager of a resource as it goes through the webhook other than manually patching the `.metadata.managedFields`. However, this is not a recommended way of setting it, per the [docs](https://kubernetes.io/docs/reference/using-api/server-side-apply/#ssa-example-configmap):
```
Caution:
The .metadata.managedFields field is managed by the API server. You should avoid updating it manually.

```

It seems whoever the manager is running the apply, is ultimately the manager for the final resource, so `kubectl-client-side-apply` or `argocd-controller` depending on how you are deploying